astropy/io/fits/column.py

```diff
@@ -2447,9 +2451,14 @@ def _convert_ascii_format(format, reverse=False):
 
         # The following logic is taken from CFITSIO:
         # For integers, if the width <= 4 we can safely use 16-bit ints for all
-        # values [for the non-standard J format code just always force 64-bit]
-        if format == 'I' and width <= 4:
-            recformat = 'i2'
+        # values, if width >= 10 we may need to accomodate 64-bit ints.
+        # [for the non-standard J format code just always force 64-bit
+        #  - why, shouldn't it be 32-bit?]
```

Comment:
Comments are not the best place to ask questions ;). Let's leave J what it is, I think we will never know why this is supported anyway.

Reply:
It was more of a rhetorical question, maybe hoping someone would someday remember what extension this was based on. The `J` format definition first entered as part of a fairly [massive merge](https://github.com/astropy/astropy/commit/346517f) by @embray, does not  look like it's possible to trace it back to the actual origin.

Reply:
Sometimes it is possible to get more information in pyfits' repository (https://github.com/spacetelescope/PyFITS/commit/9096a66efa99b5111aeaf7b3f9e2bb47a1c467f8), but here, unless Erik remembers...

Reply:
If you get desperate, I think the SVN repo is archived somewhere... but I'd rather not waste time digging if this is purely rhetorical.

Reply:
I'm trying to recall the history of the non-standard "J" format for ASCII columns.  It's been so long I can't really remember, but I believe there were some files in the wild that actually used this.  I don't remember where they came from.

Reply:
Looking at the change @pllim pointed out, its existence in the code definitely predated me working on it.  It's possible I just kept it for backwards-compatibility.  I don't recall anymore whether it's actually been used.  I wonder if anyone else knows...

Reply:
Hmm, maybe I did add it.  It first appears in the git repo at https://github.com/spacetelescope/PyFITS/commit/0e5a8a213e1f3ab919e2e577176719df3dc622cb, which according to the commit message was somehow carried over from SVN, but the commit message in git doesn't clarify further.  I might have a copy of the svn repo somewhere, let me check...